File: Python/JewelsAndStones.py
"""
Given a string representing your stones and another string representing a list of jewels, 
return the number of stones that you have that are also jewels.
"""


class JewelsAndStones:
    def numJewelsInStones(self, J: str, S: str) -> int:
        if J is None or len(J) == 0 or S is None or len(S) == 0:
            return 0
        count = 0
        for c in S:
            if c in J:
                count += 1
        return count


if __name__ == "__main__":
    j = JewelsAndStones()
    print(j.numJewelsInStones("aA", "aAAbbbb"))
    print(j.numJewelsInStones("z", "ZZ"))

File: Java/JewelsAndStones.java
/**
 * Given a string representing your stones and another string representing a
 * list of jewels, return the number of stones that you have that are also
 * jewels.
 */
public class JewelsAndStones {

    private static int numJewelsInStones(String J, String S) {
        if (J == null || J.isEmpty() || S == null || S.isEmpty()) {
            return 0;
        }
        int count = 0;
        for (char c : S.toCharArray()) {
            if (J.indexOf(c) > -1) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        System.out.println(numJewelsInStones("aA", "aAAbbbb"));
        System.out.println(numJewelsInStones("z", "ZZ"));
    }
}
